clc/modules/object-storage-common/src/main/java/com/eucalyptus/objectstorage/exceptions/s3/S3Exceptions.groovy

```diff
@@ -756,6 +845,19 @@ class NoSuchLifecycleConfigurationException extends S3Exception {
   }
 }
 
+// When no CORS configuration exists, to match AWS's response, return 404 NotFound with
+// this error code and message.
+class NoSuchCorsConfigurationException extends S3Exception {
+  def NoSuchCorsConfigurationException() {
+    super(S3ErrorCodeStrings.NoSuchCORSConfiguration, "The CORS configuration does not exist.", HttpResponseStatus.NOT_FOUND);
+  }
+
+  def NoSuchCorsConfigurationException(String resource) {
+    this();
+    this.resource = resource;
+  }
+}
+
```

Comment:
`S3Exceptions.groovy` is exclusively for S3 defined list of [error responses]( http://docs.aws.amazon.com/AmazonS3/latest/API/ErrorResponses.html#ErrorCodeList). Eucalyptus specific exceptions used internally for convenience should be defined in another class, probably under `com.eucalyptus.objectstorage.exceptions` package